pkg/reconciler/cleanup.go

```diff
@@ -36,11 +36,26 @@ func (r *Reconciler) cleanupPipelineRuns(ctx context.Context, logger *zap.Sugare
 		if err != nil {
 			return err
 		}
-
+		// if annotation value is more than max limit defined in config then use from config
+		if r.run.Info.Pac.MaxKeepRuns > 0 && max > r.run.Info.Pac.MaxKeepRuns {
+			logger.Infof("max-keep-run value in annotation is more than config, so using value from config")
+			max = r.run.Info.Pac.MaxKeepRuns
+		}
 		err = r.kinteract.CleanupPipelines(ctx, logger, repo, pr, max)
 		if err != nil {
 			return err
 		}
+		return nil
+	}
+
+	// if annotation is not defined but default max-keep-run value is defined then use that
+	if r.run.Info.Pac.DefaultMaxKeepRuns > 0 {
+		max := r.run.Info.Pac.DefaultMaxKeepRuns
+
+		err := r.kinteract.CleanupPipelines(ctx, logger, repo, pr, max)
+		if err != nil {
+			return err
+		}
```

Comment:
I see this piece of code being repeated here (first one is up [here](https://github.com/openshift-pipelines/pipelines-as-code/pull/892/files#diff-681a0da6d24af3d229eefa043b108b3da3118dd8a81f1988ae8aaab9f1ee45bfR44)).
Is this because it provides more readability?

Reply:
yeah we can move that out of the `if` cond but we don't always execute that
so if annotation is not there and default value is not set then we don't do cleanup

for that case we will again have to add a cond below 😅 due to which kept at both places

Reply:
yep, it makes sense. Thanks for explaining 😺